### PR TITLE
Fixing remaining bad links and html errors

### PR DESCRIPTION
**Pull request by ldemailly**
Fix all the issues found by html-proofer. Follow up issue #238 to check it for all changes.

Couldn't find google.rpc.Code so not a link anymore, and the quota issues has been changed in master since

Was still left :
#google.rpc.Code in reference/api/mixer/status

And the mixer-config-quota-bookinfo.yaml is missing in
https://github.com/istio/istio/tree/master/samples/apps/bookinfo

Issue #226

### CODE REVIEW

File: _docs/reference/api/mixer/mixer-service.md [hunk 1]
```diff
@@ -387,7 +387,7 @@ Used to report telemetry after performing an action.
 <a name="istio.mixer.v1.ReportResponse.result"></a>
  <tr>
   <td><code>result</code></td>
-  <td><a href="./status.status.html">Status</a></td>
```
Comment (ldemailly) [inline _docs/reference/api/mixer/mixer-service.md]: not super sure the 2 are the same/why there was a double status here originally

File: _glossary/istio-auth.md [hunk 2]
```diff
@@ -5,4 +5,4 @@ type: markdown
 {% include home.html %}
 
 Istio-Auth provides strong service-to-service and end-user authentication using mutual TLS, with built-in identity and
-credential management. Learn more about Istio-Auth [here]({{home}}/docs/concepts/network-and-auth/auth.html).
```
Comment (ldemailly) [inline _glossary/istio-auth.md]: {{home}} doesn't work in these files - somehow

File: _docs/concepts/policy-and-control/mixer-config.md [hunk 3]
```diff
@@ -452,7 +452,7 @@ manifests:
 ## Examples
 
 You can find fully formed examples of Mixer configuration by visiting the [Samples]({{home}}/docs/samples). As
-a specific example, here is the [BookInfo configuration](https://raw.githubusercontent.com/istio/istio/master/samples/mixer-config-quota-bookinfo.yaml).
```
Comment (ldemailly) [inline _docs/concepts/policy-and-control/mixer-config.md]: this file is missing in action

File: Rakefile [hunk 4]
```diff
@@ -2,6 +2,7 @@ require 'html-proofer'
 
 task :test do
   sh "bundle exec jekyll build"
-  options = { :assume_extension => true }
+  options = { :check_html => true, # :validation => { :report_missing_names, :report_invalid_tags => true, :report_script_embeds => true },
```
Comment (ijsnellf) [inline Rakefile]: Intentionally commented out?

Comment (ldemailly) [inline Rakefile]: yes because there are a ton more errors but I wanted to document those extra options

**Pull request merged**